web/packages/teleterm/src/ui/Search/SearchBar.tsx

```diff
@@ -77,8 +77,15 @@ function SearchBar() {
       return;
     }
 
-    const onClickOutside = e => {
-      if (!e.composedPath().includes(containerRef.current)) {
+    const onClickOutside = (e: MouseEvent) => {
+      if (
+        !(
+          e.composedPath().includes(containerRef.current) ||
+          // TODO(gzdunek): Remove this ugly hack that prevents closing the
+          // the search bar when the predicate tooltip is clicked (or closed)
+          document.querySelector('#predicate-documentation')
```

Comment:
> My original idea was to add `onClick={e => e.preventDefault()}` to the tooltip, so the events wouldn't propagate to the window. Unfortunately, this doesn't work because the on click handler added to the window uses `capture` flag, so the events reach that handler first.

What if that `onClick` on the toolip also uses `capture`, wouldn't it get called first and be able to call `e.preventDefault`?

https://www.robinwieruch.de/react-event-bubbling-capturing/#event-capturing-in-react

`onClickCapture` should work, though I've never used it before.

---

I don't think the way you chose to solve it is that bad, it's probably better than trying to implement `onShow(promise: Promise)` on `ToolTipInfo`, which I think would be rather convoluted, no? You'd need to persist some state between renders to resolve the promise.

Reply:
I tested that and `onClickCapture` doesn't work, the event goes first to the search bar event listener.

>trying to implement onShow(promise: Promise) on ToolTipInfo, which I think would be rather convoluted, no? You'd need to persist some state between renders to resolve the promise.

Yeah, forgot about it. In that case let's keep the current solution. I removed the TODO comment.

Reply:
> I tested that and `onClickCapture` doesn't work, the event goes first to the search bar event listener.

That's unfortunate. I didn't really know what is the order for those events. It looks like [capturing goes top to bottom](https://javascript.info/bubbling-and-capturing#capturing), so indeed the window listener would be called first.